cmd/jx/main.go

```diff
@@ -5,34 +5,51 @@ import (
 	"os"
 
 	"foxygo.at/jsonnext"
+	jxkong "foxygo.at/jsonnext/kong"
 	"github.com/alecthomas/kong"
 	"github.com/google/go-jsonnet"
 )
 
-var CLI struct {
-	ImportPath []string `name:"jpath" sep:":" short:"J"`
-	Filename   string   `arg:"" optional:"" help:"File to evaluate. stdin is used if omitted or \"-\""`
+type config struct {
+	jsonnext.Config
+	Filename string `arg:"" optional:"" help:"File to evaluate. stdin is used if omitted or \"-\""`
 }
 
 func main() {
-	kong.Parse(&CLI)
+	cli := parseCLI()
 
-	vm := jsonnet.MakeVM()
-	i := newImporter(CLI.ImportPath)
-	vm.Importer(i)
+	vm := newVM(&cli.Config)
 
-	out, err := run(vm, CLI.Filename)
+	out, err := run(vm, cli.Filename)
 	if err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
 	fmt.Print(out)
 }
 
-func newImporter(importPath []string) *jsonnext.Importer {
-	i := jsonnext.Importer{SearchPath: importPath}
-	i.AppendSearchFromEnv("JXPATH")
-	return &i
+func parseCLI() *config {
+	c := &config{}
+	c.Config = *jsonnext.NewConfig()
+	cli := struct {
+		Config *config        `embed:""`
+		Kong   *jxkong.Config `embed:""`
+	}{
+		Config: c,
+		Kong:   jxkong.NewConfig(&c.Config),
```

Comment:
Done. Please give your feedback on how it now works.

Reply:
yeah, nice, I like this a lot.

Reply:
I guess the cli.Config.Config is the only remaining imperfection. Just can't have it all I guess.

Reply:
That has actually been cleaned up a bit with your `MakeVM()` suggestion. Because that is a method, it is "inherited" due to the embedding, so `cli.Config.MakeVM()` works without needing the extra `.Config`